source/game.mission.highscore.bmx

```diff
@@ -0,0 +1,97 @@
+SuperStrict
+
+Import "Dig/base.util.data.bmx"
+?bmxng
+Import "Dig/external/persistence.mod/persistence_mxml.bmx"
+?not bmxng
+Import "Dig/external/persistence.mod/persistence.bmx"
+?
+
+Type TMissionHighscore
+	'TODO how to access the local time for displaying when the game was played
+	'Field realDate:Long
+	Field playerID:Int
+	Field playerName:String
+	Field channelName:String
+	Field gameMinutes:Long
+	Field missionAccomplished:Int = False
+	Field aiPlayer:Int = False
+	Field playerDifficulty:String
```

Comment:
Die "Difficulty" rein vom Namen her zu persistieren - das nuetzt nur was, wenn sich bspweise zwischen TVT-Versionen nichts aendert.

- Spielversion mit speichern
- Ueberlegen ob man dem Schwierigkeitsgrad einen "Hash" mit gibt (zur not einfach alle Difficultywerte "ver-string-en", dann "GesamtString.Hash()" abrufen)
- oder ob der Schwierigkeitsgrad irgendwie komplett persistiert werden sollte
- oder Schwierigkeitsgrade bekommen "Revisionsnummern" ... sozusagen "Schwer Rev.1246" - was aber nur was bringt, wenn der Spieler an die dazugehoerigen Informationen rankommt

Natuerlich ist es am Einfachsten, einfach eine fixe Konfiguration der Schwierigkeitsgrade anzunehmen - dann kann man es einfach so belassen, wie Du das umgesetzt hast.

Reply:
Genau die Annahme fixer Konfigurationen mache ich momentan. Die Spielerschwierigkeitsgrade sind ja nicht nur durch Versionswechsel anpassbar, sondern lokal auch vom Spieler konfigurierbar. Eine globale Vergleichbarkeit lässt sich da nicht wirklich herstellen. Und wenn die "zugehörigen" Highscores verschwinden, wenn wir den Basiskredit von 25000 auf 24999 anpassen, wird auch niemanden glücklich machen. Die zusätzliche Information über Revision/Hash/komplette Konfiguration am Spielstand macht die Sache auch nicht übersichtlicher (insb. wenn man cheaten will und die Angaben erst kurz vor Missionserreichen im Spielstand anpasst).

Ich habe den Spielerschwierigkeitsgrad integriert (informativer Charakter z.B. wenn eine KI mit einem vom Spieler abweichenden Level auftaucht), da wir möglicherweise nicht in allen Missionen die Werte vordefinieren werden. Aktuell ist die Information "redundant", da das Level vom Missionslevel bestimmt wird.